linalg/operator.hpp

```diff
@@ -363,6 +363,27 @@ class TimeDependentOperator : public Operator
        If not re-implemented, this method simply generates an error. */
    virtual void ImplicitSolve(const double dt, const Vector &x, Vector &k);
 
+   /** @brief Solve the equation: @a k = f(@a x + 1/2 @a dt0^2 @a k, @a dxdt + @a dt1 @a k, t), for the
```

Comment:
this is my only concern about this PR. I think it is conceptually wrong to add this method to  `TimeDependentOperator`. These second order schemes should only be available when using a different class, say `SecondOrderTimeDependentOperator` (or a similar name) @tzanio?

Reply:
I had a separate class initially, but it created a lot of duplicate code regarding all the related infrastructure. Get/SetTime, Implicit/Explicite checks etc. So I see your point, but I think this is the "easiest" way..

Given that there are separate function calls in timedependentoperator for Petsc and Sundials Mass solver (for example)I believe this is not a larger intrusion on the API.

Reply:
What do you mean by "a lot"? this is just declaring methods of the class and it is how C++ is supposed to work.
You should not be afraid of writing the proper abstraction and take the "easiest" path only for this specific reason. 

PETSc (I don't know the details of Sundials) need a special callback for the Jacobian setup which is general. Instead, the MFEM interface only asks for solve the nonlinear system at onw specific point of time (given u and dudt)

Reply:
I will go the inherit route, I will update today or tomorrow.

Reply:
I have commit the changes. It might need some polishing. `SecondOrderTimeDependentOperator` is quite long.